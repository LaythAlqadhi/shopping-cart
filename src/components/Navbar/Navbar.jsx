import { Link } from 'react-router-dom';
import css from './Navbar.module.css';

const TopNavbar = () => {
  return (
    <nav className={css.nav}>
      <ul>
        <li className={css.logo}>
          <Link to="/">
            <h1 className="logo">
              eShop
            </h1>
          </Link>
        </li>
        <li>
          <Link to="/" className="material-symbols-outlined">
            home
          </Link>
        </li>
        <li>
          <Link to="shop" className="material-symbols-outlined">
            store
          </Link>
        </li>
        <li>
          <Link to="cart" className="material-symbols-outlined">
            shopping_cart
          </Link>
        </li>
      </ul>
    </nav>
  );
}

export default TopNavbar;