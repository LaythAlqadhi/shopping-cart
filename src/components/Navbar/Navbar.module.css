.nav {
  width: 100%;
  padding: 10px;
  background-color: var(--primary-accent-color);
}

.nav ul {
  width: 100%;
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.nav ul li {
  list-style: none;
  color: var(--primary-color);
  margin-inline: 10px;
}

.nav ul li a {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: var(--primary-color);
}

.nav ul .logo {
  margin-right: auto;
}