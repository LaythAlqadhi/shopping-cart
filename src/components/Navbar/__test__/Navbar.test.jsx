import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import Navbar from '../Navbar';

describe('Navbar component', () => {
  it('renders in the document', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <Navbar />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
})