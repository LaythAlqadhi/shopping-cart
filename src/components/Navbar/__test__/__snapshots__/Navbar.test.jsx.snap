// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar component > renders in the document 1`] = `
<div>
  <nav
    class="_nav_278c00"
  >
    <ul>
      <li
        class="_logo_278c00"
      >
        <a
          href="/"
        >
          <h1
            class="logo"
          >
            eShop
          </h1>
        </a>
      </li>
      <li>
        <a
          class="material-symbols-outlined"
          href="/"
        >
          home
        </a>
      </li>
      <li>
        <a
          class="material-symbols-outlined"
          href="/shop"
        >
          store
        </a>
      </li>
      <li>
        <a
          class="material-symbols-outlined"
          href="/cart"
        >
          shopping_cart
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Navbar component > renders in the document with three link components 1`] = `
<div>
  <nav
    class="_nav_278c00"
  >
    <ul>
      <li
        class="_logo_278c00"
      >
        <a
          href="/"
        >
          <h1
            class="logo"
          >
            eShop
          </h1>
        </a>
      </li>
      <li>
        <a
          class="material-symbols-outlined"
          href="/"
        >
          home
        </a>
      </li>
      <li>
        <a
          class="material-symbols-outlined"
          href="/shop"
        >
          store
        </a>
      </li>
      <li>
        <a
          class="material-symbols-outlined"
          href="/cart"
        >
          shopping_cart
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
